### ISSUE TITLE
wc_format_phone_number: bad format

### ISSUE THREAD

**Issue by lwxbr** [woocommerce/woocommerce#23947]
**Describe the bug**

The function **wc_format_phone_number** is formatting for example, in Brazil, a normal phone number would be (32) 3212-2345. The function is formatting to -32- 3212-2345. The regular expression is not ignoring the parens ( "(" and ")" ). There are some workaround? My opinion to fix this issue is create a hook to overwite the format I want and even create a pattern of masks to format the fields.

**To Reproduce**
1. Go to 'my account'
2. Click on 'addresses'
3. Edit an address and fill the phone using this format (00) 0000-0000 (where 0 is a number in interval 0-9)
4. See error: -00- 0000-0000

**Screenshots**
![image](https://user-images.githubusercontent.com/40926976/59614089-6c9b9180-90f6-11e9-9b4c-4ce5c06c476c.png)

**Expected behavior**
Just create a hook to customize the format I want to use. That's fine and that will solve future issues envolving phone formatting.

**Isolating the problem (mark completed items with an [x]):**
- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**WordPress Environment**

```
`
### WordPress Environment ###

Home URL: https://localhost/market
Site URL: https://localhost/market
WC Version: 3.4.7
Log Directory Writable: ✔
WP Version: ❌ 4.9.10 - Há uma nova versão do WordPress disponível (5.2.1)
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: pt_BR
External object cache: –

### Server Environment ###

Server Info: Apache/2.4.37 (Win32) OpenSSL/1.1.1a PHP/7.3.1
PHP Version: 7.3.1
PHP Post Max Size: 128 MB
PHP Time Limit: 300
PHP Max Input Vars: 1000
cURL Version: 7.61.1
OpenSSL/1.1.1a

SUHOSIN Installed: –
MySQL Version: 5.5.5-10.1.37-MariaDB
Max Upload Size: 64 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.4.7
WC Database Prefix: wp_
MaxMind GeoIP Database: ✔
Tamanho total do banco de dados: 2.70MB
Tamanho dos dados do banco de dados: 1.72MB
Tamanho do índice do banco de dado: 0.98MB
wp_woocommerce_sessions: Dados: 0.02MB + Índices: 0.03MB
wp_woocommerce_api_keys: Dados: 0.02MB + Índices: 0.03MB
wp_woocommerce_attribute_taxonomies: Dados: 0.02MB + Índices: 0.02MB
wp_woocommerce_downloadable_product_permissions: Dados: 0.02MB + Índices: 0.05MB
wp_woocommerce_order_items: Dados: 0.02MB + Índices: 0.02MB
wp_woocommerce_order_itemmeta: Dados: 0.02MB + Índices: 0.03MB
wp_woocommerce_tax_rates: Dados: 0.02MB + Índices: 0.06MB
wp_woocommerce_tax_rate_locations: Dados: 0.02MB + Índices: 0.03MB
wp_woocommerce_shipping_zones: Dados: 0.02MB + Índices: 0.00MB
wp_woocommerce_shipping_zone_locations: Dados: 0.02MB + Índices: 0.03MB
wp_woocommerce_shipping_zone_methods: Dados: 0.02MB + Índices: 0.00MB
wp_woocommerce_payment_tokens: Dados: 0.02MB + Índices: 0.02MB
wp_woocommerce_payment_tokenmeta: Dados: 0.02MB + Índices: 0.03MB
wp_woocommerce_log: Dados: 0.02MB + Índices: 0.02MB
wp_commentmeta: Dados: 0.02MB + Índices: 0.03MB
wp_comments: Dados: 0.02MB + Índices: 0.09MB
wp_links: Dados: 0.02MB + Índices: 0.02MB
wp_options: Dados: 1.05MB + Índices: 0.02MB
wp_postmeta: Dados: 0.09MB + Índices: 0.06MB
wp_posts: Dados: 0.06MB + Índices: 0.06MB
wp_termmeta: Dados: 0.02MB + Índices: 0.03MB
wp_terms: Dados: 0.02MB + Índices: 0.03MB
wp_term_relationships: Dados: 0.02MB + Índices: 0.02MB
wp_term_taxonomy: Dados: 0.02MB + Índices: 0.03MB
wp_usermeta: Dados: 0.02MB + Índices: 0.03MB
wp_users: Dados: 0.02MB + Índices: 0.05MB
wp_wc_download_log: Dados: 0.02MB + Índices: 0.03MB
wp_wc_product_meta_lookup: Dados: 0.02MB + Índices: 0.09MB
wp_wc_webhooks: Dados: 0.02MB + Índices: 0.02MB

### Post Type Counts ###

attachment: 17
customize_changeset: 1
page: 8
post: 2
product: 12
revision: 2
scheduled-action: 19
shop_coupon: 1
shop_order: 5
shop_order_refund: 1

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (3) ###

WooCommerce: por Automattic – 3.4.7 – 3.6.4 está disponível
WP Reset: por WebFactory Ltd – 1.60
WP Rollback: por Impress.org – 1.6

### Settings ###

API Enabled: –
Force SSL: –
Currency: BRL (R$)
Currency Position: left
Thousand Separator: .
Decimal Separator: ,
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)


### WC Pages ###

Base da loja: #5 - /loja/
Carrinho: #6 - /carrinho/
Finalizar compra: #7 - /finalizar-compra/
Minha conta: #8 - /minha-conta/
Termos e condições: ❌ Página não definida

### Theme ###

Name: Storefront
Version: 2.5.0
Author URL: https://woocommerce.com/
Child Theme: ❌ – Recomendamos você utilizar um tema filho no caso de você estiver personalizando o WooCommerce em um tema ascendente. Veja como criar um tema filho
WooCommerce Support: ✔

### Templates ###

Overrides: –
`
```

**Comment by claudiosanches** [comment]
I'm not able to reproduce in the way you describable. You are using an outdated version of WooCommerce, this has already been changed and this problem doesn't happen on the screen that you pointed out, still is possible when using the `wc_format_phone_number()` function.

I recommend you to update your WooCommerce installation, any way a fix for this function should be included into WooCommerce 3.6.5 anyway.

**Comment by claudiosanches** [comment]
Thank you for reporting this issue!

I'm closing the issue as it already has a PR, please continue the discussion over at #23967.
